From 9a332b273326d2e01b00cf31d32d7737a9590eee Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Fri, 4 Jul 2014 15:26:41 +0800
Subject: [PATCH] ENGR00321285 snvs_pwrkey: default timeout for snvs power key
 is too short(5s)

Make cold shutdown timeout to 15s for snvs power keyboard

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 drivers/input/keyboard/snvs_pwrkey.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/input/keyboard/snvs_pwrkey.c b/drivers/input/keyboard/snvs_pwrkey.c
index 1d6d62c..b16d4a8 100644
--- a/drivers/input/keyboard/snvs_pwrkey.c
+++ b/drivers/input/keyboard/snvs_pwrkey.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2011-2014 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2011-2015 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * The code contained herein is licensed under the GNU General Public
  * License. You may obtain a copy of the GNU General Public License
@@ -28,6 +28,10 @@
 #define SNVS_HPSR_BTN	(0x1 << 6)
 #define SNVS_LPSR_SPO	(0x1 << 18)
 #define SNVS_LPCR_DEP_EN (0x1 << 5)
+#define SNVS_LPCR_BTN_PRESSTIME_5S (0x0 << 16)
+#define SNVS_LPCR_BTN_PRESSTIME_10S (0x1 << 16)
+#define SNVS_LPCR_BTN_PRESSTIME_15S (0x2 << 16)
+#define SNVS_LPCR_BTN_PRESSTIME_DISABLE (0x3 << 16)
 
 struct pwrkey_drv_data {
 	void __iomem *ioaddr;
@@ -116,7 +120,9 @@ static int imx_snvs_pwrkey_probe(struct platform_device *pdev)
 
 	ioaddr = pdata->ioaddr;
 	val = readl_relaxed(ioaddr + SNVS_LPCR_REG);
-	val |= SNVS_LPCR_DEP_EN,
+	val |= SNVS_LPCR_DEP_EN;
+	val &= ~SNVS_LPCR_BTN_PRESSTIME_DISABLE;
+	val |= SNVS_LPCR_BTN_PRESSTIME_15S;
 	writel_relaxed(val, ioaddr + SNVS_LPCR_REG);
 	/* clear the unexpected interrupt before driver ready */
 	val = readl_relaxed(ioaddr + SNVS_LPSR_REG);
-- 
1.8.0

